scripts/pylib/twister/twisterlib/testplan.py

```diff
@@ -549,6 +554,46 @@ def add_testsuites(self, testsuite_filter=None):
                                 suite.platform_allow,
                                 f"platform_allow in {suite.name}")
 
+                        if self.options.log_level == "QA":
+                            if len(suite.platform_allow) > 10:
+                                logger.qa(f"Too many entries in allow_platform in {suite.name}: {len(suite.platform_allow)}")
+
+                            if len(suite.integration_platforms) > 5:
+                                logger.qa(f"Too many integration platforms in {suite.name}: {len(suite.integration_platforms)}")
+
+                            if len(suite.platform_exclude) > 5:
+                                logger.qa(f"Too many excluded platforms in {suite.name}: {len(suite.platform_exclude)}")
+
+                            if len(suite.platform_allow) > 1 and not suite.integration_platforms:
+                                logger.qa(f"platform_allow in {suite.name} without integration_platforms")
+
+
+                            if suite.build_only and not "build" in suite.id:
+                                logger.qa(f"build_only set in {suite.name}")
+
+                            if suite.skip:
+                                logger.qa(f"skip set in {suite.name}")
+
+                            if suite.harness not in SUPPORTED_HARNESS:
+                                logger.qa(f"Unsupported harness {suite.harness} in {suite.name}")
+
+                            if suite.filter and not suite.integration_platforms and not suite.platform_allow:
+                                logger.qa(f"filter with no integration platforms in {suite.name}")
```

Comment:
the condition also depends on platform_allow. Why? The printed output doesn't match this.